tests/tasks/assert_fail_on_s390x.yml

```diff
@@ -0,0 +1,13 @@
+# SPDX-License-Identifier: MIT
+---
+- name: Assert failure on the s390x architecture
+  when: ansible_architecture == 's390x'
```

Comment:
You want the `assert:` task to run on all platforms.  Since this code is in the `rescue` block, you want the test to fail if there is an error and it isn't `This role does not support s390x architecture`.

Also, I don't think `end_host` will skip the `always` block

Reply:
I actually want to assert to run only on s390x. Other archs in the case of failure should skip this to avoid additional noise on the output.
end_host does a hard finish of a host. From a tests against s390x:

```
TASK [linux-system-roles.bootloader : Fail on s390x architecture] **************
fatal: [s390x]: FAILED! => {"changed": false, "msg": "This role does not support s390x architecture. Let us know if you need the role to support it."}

TASK [Assert failure on s390x architecture] ************************************
included: /home/spetrosi/Documents/work/linux-system-roles/bootloader/tests/tasks/assert_fail_on_s390x.yml for s390x

TASK [Assert failure] **********************************************************
ok: [s390x] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [End execution on s390x host to avoid failures in clean up] ***************

PLAY RECAP *********************************************************************
s390x                      : ok=6    changed=0    unreachable=0    failed=0    skipped=1    rescued=1    ignored=0   
```

Reply:
re: `always` - that's good to know - not sure if there are assumptions in other roles that `always` will run for `end_host` or `end_play`

re: `rescue` - what happens if the role fails on x86_64?  I think the `rescue` will catch the error, and will continue, and the test will report success.

Reply:
re: rescue, that's right. I am fixing this in a new commit,